examples/Model/User.php

```diff
@@ -43,9 +43,9 @@ function changeUserName(array $state, ChangeUserName $command): AggregateResult
         throw new InvalidArgumentException('Username too short');
     }
 
-    $event = new UserNameWasChanged($command->payload());
+    $event = new UserNameWasChanged($command->payload(), ['_aggregate_version' => $state['version'] + 1]);
```

Comment:
apply would be wrong. maybe a helper function to create events which requires the version as argument?